htdocs/api/v0.0.1/candidates/Visit.php

```diff
@@ -0,0 +1,195 @@
+                                  'Visit'   => $this->VisitLabel,
+                                  'Battery' => $SubProjTitle,
+                                 ],
+                      ];
+        if ($this->Timepoint) {
+            $stages = [];
+            if ($this->Timepoint->getDateOfScreening() !== null) {
+                $Date   = $this->Timepoint->getDateOfScreening();
+                $Status = $this->Timepoint->getScreeningStatus();
+
+                $stages['Screening'] = [
+                                        'Date'   => $Date,
+                                        'Status' => $Status,
+                                       ];
+            }
+            if ($this->Timepoint->getDateOfVisit() !== null) {
+                $Date   = $this->Timepoint->getDateOfVisit();
+                $Status = $this->Timepoint->getVisitStatus();
+
+                $stages['Visit'] = [
+                                    'Date'   => $Date,
+                                    'Status' => $Status,
+                                   ];
+            }
+            if ($this->Timepoint->getDateOfApproval() !== null) {
+                $Date   = $this->Timepoint->getDateOfApproval();
+                $Status = $this->Timepoint->getApprovalStatus();
+
+                $stages['Approval'] = [
+                                       'Date'   => $Date,
+                                       'Status' => $Status,
+                                      ];
+            }
+            $this->JSON['Stages'] = $stages;
+        }
+    }
+
+    /**
+     * Handles a PUT request for a visit
+     *
+     * @return none
+     */
+    public function handlePUT()
+    {
+        if (!isset($this->ReceivedJSON['Meta']['CandID'])
+            || $this->ReceivedJSON['Meta']['CandID'] != $this->CandID
+        ) {
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->error("Candidate from URL does not match metadata");
+                $this->safeExit(0);
+        }
+        if (!isset($this->ReceivedJSON['Meta']['Visit'])
+            || $this->ReceivedJSON['Meta']['Visit'] != $this->VisitLabel
+        ) {
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->error("Visit from URL does not match metadata");
+                $this->safeExit(0);
+        }
+
+        $subprojects  = \Utility::getSubprojectList();
+        $subprojectID = null;
+        foreach ($subprojects as $subproject => $title) {
+            if ($title === $this->ReceivedJSON['Meta']['Battery']) {
+                $subprojectID = $subproject;
+                break;
+            }
+        }
+        if ($subprojectID === null) {
+            $this->header("HTTP/1.1 400 Bad Request");
+            $this->error("Test battery specified does not exist");
+            $this->safeExit(0);
+
+        }
+        // need to extract subprojectID
+        //TimePoint::createNew($this->CandID, $subprojectID, $this->VisitLabel);
```

Comment:
is this supposed to be commented out, if not should it also be set to the root ```\``` space